lib/TopStoriesFeed.jsm

```diff
@@ -30,17 +30,29 @@ const SPOC_IMPRESSION_TRACKING_PREF = "feeds.section.topstories.spoc.impressions
 const REC_IMPRESSION_TRACKING_PREF = "feeds.section.topstories.rec.impressions";
 const OPTIONS_PREF = "feeds.section.topstories.options";
 const MAX_LIFETIME_CAP = 500; // Guard against misconfiguration on the server
+const DISCOVERY_STREAM_PREF = "discoverystream.config";
 
 this.TopStoriesFeed = class TopStoriesFeed {
-  constructor() {
-    this.spocCampaignMap = new Map();
+  constructor(ds) {
+    this.discoveryStreamEnabled = ds && ds.value && JSON.parse(ds.value).enabled;
+    if (!this.discoveryStreamEnabled) {
+      this.initializeProperties();
+    }
```

Comment:
@Mardak With this check in lazyLoadingTopStories, do you think we need it in constructor?

Reply:
Thinking more we should keep default initialization in place for TopStoriesFeed API

Reply:
The general pattern for feeds is to initialize properties that might be used from entrypoints (e.g., action callbacks) while deferring any actual work to INIT or other actions. If we do move properties to a later time, are we confident no entrypoint will try to access them? E.g., on SYSTEM_TICK, does it try to check some property that no longer exists?